fairroot/base/steer/FairRunAna.cxx

```diff
@@ -551,6 +551,7 @@ void FairRunAna::TerminateRun()
 {
     fRootManager->StoreAllWriteoutBufferData();
     fTask->FinishTask();
+    GetSource()->Finish();
```

Comment:
And this one is the one where I wonder, whether Close is called at all.

This is in `FairRunAna::TerminateRun`. And the only place I could find that calls into `TerminateRun` is in FairAnaSelector, which is only used in FairRunAnaProof. So it looks like this is never called. I again wonder, where `GetSource()->Close()` gets really called?

Reply:
The TerminateRun is never called (see this issue #1462)

GetSource()->Close() is only called in FairRunOnline::Finish(), which is called in FairRun::Run(). It won't be called neither if Run() is skipped.

Reply:
Let's take a deeper look at #1462 first, please!